### PR TITLE
Assignable PRs

### PR DESCRIPTION
**Pull request by Dantemss**
This PR is based on the production version and includes dependabot PRs and stuff needed for assignable

### CODE REVIEW

**Comment by Dantemss** [comment]
The assorted commits are from 2 PRs that I had normal-merged instead of squash-merged.

**Comment by Dantemss** [comment]
Commits 8484508 and newer contain new code.

File: app/routines/find_or_create_user.rb [hunk 1]
```diff
@@ -18,19 +18,33 @@ class FindOrCreateUnclaimedUser
...
     # output either the found unclaimed user or a freshly created one
-    outputs[:user] = user || create_user(options)
+    outputs.user = find_user(options) || create_user(options)
+  end
+
+  # Attempt to find a user by either the external_id, username, or email address
+  def find_user(options)
+    if options[:external_id].present?
+      User.joins(:external_ids).find_by(external_ids: { external_id: options[:external_id] })
+    elsif options[:username].present?
```
Comment (mwvolo) [inline app/routines/find_or_create_user.rb:29]: are you aware of any applications still using username?

Comment (Dantemss) [reply app/routines/find_or_create_user.rb:29]: No, not that I know of

**Review by mwvolo — APPROVED** [review]

**Pull request merged**